### PR TITLE
ftplugin(dts): add basic support

### PR DESCRIPTION
**Pull request by Freed-Wu**

### CODE REVIEW

File: runtime/ftplugin/dts.vim [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+" Language:             dts
+" Maintainer:           Wu, Zhenyu <wuzhenyu@ustc.edu>
+" Latest Revision:      2024-04-12
+
+if exists('b:did_ftplugin')
+  finish
+endif
+let b:did_ftplugin = 1
+
+let s:save_cpoptions = &cpoptions
```
Comment (chrisbra) [inline runtime/ftplugin/dts.vim]: that is not needed

**Review by chrisbra — CHANGES_REQUESTED** [review]

File: runtime/ftplugin/dts.vim (hunk 1 above)
Comment (Freed-Wu) [reply runtime/ftplugin/dts.vim]: Fixed.

File: runtime/ftplugin/dts.vim [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+" Language:             dts
+" Maintainer:           Wu, Zhenyu <wuzhenyu@ustc.edu>
+" Latest Revision:      2024-04-12
+
+if exists('b:did_ftplugin')
+  finish
+endif
+let b:did_ftplugin = 1
+
+let b:undo_ftplugin = 'setl <inc cms< com<'
```
Comment (dkearns) [inline runtime/ftplugin/dts.vim]:
```suggestion
let b:undo_ftplugin = 'setl inc< cms< com<'
```

File: runtime/ftplugin/dts.vim [hunk 3]
```diff
@@ -0,0 +1,16 @@
+" Vim filetype plugin file
+" Language:             dts
```
Comment (dkearns) [inline runtime/ftplugin/dts.vim]:
```suggestion
" Language:     dts/dtsi (device tree files)
```

This matches the syntax file header.

File: runtime/ftplugin/dts.vim [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+" Maintainer:           Wu, Zhenyu <wuzhenyu@ustc.edu>
+" Latest Revision:      2024-04-12
```
Comment (dkearns) [inline runtime/ftplugin/dts.vim]:
```suggestion
" Last Change:      2024 Apr 12
```

I know there are examples of the other header format.  I'll update them.

**Review by dkearns — CHANGES_REQUESTED** [review]

File: runtime/ftplugin/dts.vim (hunk 2 above)
Comment (Freed-Wu) [reply runtime/ftplugin/dts.vim]: Done.

File: runtime/ftplugin/dts.vim (hunk 4 above)
Comment (Freed-Wu) [reply runtime/ftplugin/dts.vim]: Done.

File: runtime/ftplugin/dts.vim (hunk 3 above)
Comment (Freed-Wu) [reply runtime/ftplugin/dts.vim]: Done.

**Comment by chrisbra** [comment]
thanks!

**Pull request merged**